src/stringProcessing/createRegexFromArray.js

```diff
@@ -1,28 +1,39 @@
 /** @module stringProcessing/createRegexFromArray */
 
-import addWordBoundary from "../stringProcessing/addWordboundary.js";
-
-import { map } from "lodash-es";
+import { map, memoize } from "lodash-es";
+import addWordBoundary from "./addWordboundary.js";
+import replaceDiacritics from "./replaceDiacritics";
+import sanitizeString from "./sanitizeString";
 
 /**
  * Creates a regex of combined strings from the input array.
  *
  * @param {array} array The array with strings
- * @param {boolean} [disableWordBoundary] Boolean indicating whether or not to disable word boundaries.
+ * @param {boolean} [disableWordBoundary=false] Boolean indicating whether or not to disable word boundaries.
+ * @param {string} [extraBoundary=""] A string that is used as extra boundary for the regex.
+ * @param {boolean} [doReplaceDiacritics=false] If set to true, it replaces diacritics. Defaults to false.
+ *
  * @returns {RegExp} regex The regex created from the array.
  */
-export default function( array, disableWordBoundary ) {
-	var regexString;
-	var _disableWordBoundary = disableWordBoundary || false;
+export default memoize( function( array, disableWordBoundary = false, extraBoundary = "", doReplaceDiacritics = false ) {
```

Comment:
With `memoize` by default only the first argument is used (https://lodash.com/docs/4.17.11#memoize). `array` could be a new reference every time. Therefore, either a resolver should be made, or `memoize` should not be used here.